daemon/update.go

```diff
@@ -40,22 +40,27 @@ func (daemon *Daemon) update(name string, hostConfig *container.HostConfig) erro
 		return err
 	}
 
-	restoreConfig := false
 	backupHostConfig := *ctr.HostConfig
+
+	restoreConfig := false
 	defer func() {
 		if restoreConfig {
 			ctr.Lock()
-			ctr.HostConfig = &backupHostConfig
-			ctr.CheckpointTo(daemon.containersReplica)
+			if !ctr.RemovalInProgress && !ctr.Dead {
```

Comment:
Should this use `restoreConfig` instead? (looks like that variable is for that purpose)

```suggestion
			if restoreConfig {
```

Reply:
No, at least because restoreConfig is set without container lock in line https://github.com/moby/moby/blob/1e1b8bf88ab03bb7dbdfb9b44039b1ba5c0090e1/daemon/update.go#L88.  As a result it possible what the restoreConfig will be true when container is dead and/or container removing will be activated.

Reply:
oh 🤦 I see this whole block is already in a `if  restoreConfig {` block. Ignore me